[2025-04-24T12:18:27.949+0000] {local_task_job_runner.py:120} INFO - ::group::Pre task execution logs
[2025-04-24T12:18:27.977+0000] {taskinstance.py:2076} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: test_spark_connection.run_simple_spark_job manual__2025-04-24T12:16:28.525029+00:00 [queued]>
[2025-04-24T12:18:27.983+0000] {taskinstance.py:2076} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: test_spark_connection.run_simple_spark_job manual__2025-04-24T12:16:28.525029+00:00 [queued]>
[2025-04-24T12:18:27.984+0000] {taskinstance.py:2306} INFO - Starting attempt 2 of 3
[2025-04-24T12:18:27.996+0000] {taskinstance.py:2330} INFO - Executing <Task(SparkSubmitOperator): run_simple_spark_job> on 2025-04-24 12:16:28.525029+00:00
[2025-04-24T12:18:28.003+0000] {standard_task_runner.py:63} INFO - Started process 479 to run task
[2025-04-24T12:18:28.005+0000] {standard_task_runner.py:90} INFO - Running: ['airflow', 'tasks', 'run', 'test_spark_connection', 'run_simple_spark_job', 'manual__2025-04-24T12:16:28.525029+00:00', '--job-id', '193', '--raw', '--subdir', 'DAGS_FOLDER/test_spark_connection.py', '--cfg-path', '/tmp/tmp9rrvo8ee']
[2025-04-24T12:18:28.006+0000] {standard_task_runner.py:91} INFO - Job 193: Subtask run_simple_spark_job
[2025-04-24T12:18:28.038+0000] {task_command.py:426} INFO - Running <TaskInstance: test_spark_connection.run_simple_spark_job manual__2025-04-24T12:16:28.525029+00:00 [running]> on host 5b95d4068e09
[2025-04-24T12:18:28.098+0000] {taskinstance.py:2648} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='airflow' AIRFLOW_CTX_DAG_ID='test_spark_connection' AIRFLOW_CTX_TASK_ID='run_simple_spark_job' AIRFLOW_CTX_EXECUTION_DATE='2025-04-24T12:16:28.525029+00:00' AIRFLOW_CTX_TRY_NUMBER='2' AIRFLOW_CTX_DAG_RUN_ID='manual__2025-04-24T12:16:28.525029+00:00'
[2025-04-24T12:18:28.099+0000] {taskinstance.py:430} INFO - ::endgroup::
[2025-04-24T12:18:28.120+0000] {base.py:84} INFO - Using connection ID 'spark_default' for task execution.
[2025-04-24T12:18:28.121+0000] {spark_submit.py:474} INFO - Spark-Submit cmd: spark-submit --master ef77e27ac9ba:7077 --name arrow-spark --verbose --deploy-mode cluster ./Scripts/test.py arg1 arg2
[2025-04-24T12:18:28.215+0000] {spark_submit.py:644} INFO - /home/***/.local/lib/python3.12/site-packages/pyspark/bin/load-spark-env.sh: line 68: ps: command not found
[2025-04-24T12:18:28.216+0000] {spark_submit.py:644} INFO - JAVA_HOME is not set
[2025-04-24T12:18:28.216+0000] {taskinstance.py:441} INFO - ::group::Post task execution logs
[2025-04-24T12:18:28.221+0000] {taskinstance.py:2905} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 465, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 432, in _execute_callable
    return execute_callable(context=context, **execute_callable_kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/baseoperator.py", line 401, in wrapper
    return func(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/providers/apache/spark/operators/spark_submit.py", line 197, in execute
    self._hook.submit(self.application)
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/providers/apache/spark/hooks/spark_submit.py", line 566, in submit
    raise AirflowException(
airflow.exceptions.AirflowException: Cannot execute: spark-submit --master ef77e27ac9ba:7077 --name arrow-spark --verbose --deploy-mode cluster ./Scripts/test.py arg1 arg2. Error code is: 1.
[2025-04-24T12:18:28.227+0000] {taskinstance.py:1206} INFO - Marking task as UP_FOR_RETRY. dag_id=test_spark_connection, task_id=run_simple_spark_job, run_id=manual__2025-04-24T12:16:28.525029+00:00, execution_date=20250424T121628, start_date=20250424T121827, end_date=20250424T121828
[2025-04-24T12:18:28.237+0000] {standard_task_runner.py:110} ERROR - Failed to execute job 193 for task run_simple_spark_job (Cannot execute: spark-submit --master ef77e27ac9ba:7077 --name arrow-spark --verbose --deploy-mode cluster ./Scripts/test.py arg1 arg2. Error code is: 1.; 479)
[2025-04-24T12:18:28.257+0000] {local_task_job_runner.py:240} INFO - Task exited with return code 1
[2025-04-24T12:18:28.268+0000] {taskinstance.py:3503} INFO - 0 downstream tasks scheduled from follow-on schedule check
[2025-04-24T12:18:28.268+0000] {local_task_job_runner.py:222} INFO - ::endgroup::
